src/MathOptInterface.jl

```diff
@@ -33,14 +33,26 @@ function optimize! end
 
 """
     write_to_file(model::ModelLike, filename::String)
+    write_to_file(model::ModelLike, io::IO)
```

Comment:
This isn't well defined. `write_to_file` extracts the file format from the file extension. An `IO` object doesn't specify the file format.

Reply:
see discussion in https://github.com/odow/MathOptFormat.jl/issues/39
and new function at https://github.com/odow/MathOptFormat.jl/pull/40/files#diff-8da2c3201bdc254700cc0b2b20f2fc8eR26
it "works" in MOF tests because the model type itself specifies the file type
@odow thoughts?

Reply:
It seems like `write_to_file` and `read_from_file` should only ever be defined for concrete `model` arguments. Depending on parsing the format from the extension is just looking for trouble.

Alternatively, we could have something like:
```julia
module MOI

abstract type AbstractFileType end

"""
    write_to_file(model::ModelLike, filename::String)
    write_to_file(model::ModelLike, io::IO, file_type::AbstractFileType)
"""
function write_to_file end

end
```
```julia
module MathOptFormat

struct MOFFile <: MOI.AbstractFileType
struct MOFModel <: MOI.ModelLike end

function MOI.write_to_file(model::MOFModel, io, ::MOFFile)
    # ...
end

function MOI.write_to_file(model::MOFModel, filename)
    if endswith(filename, ".mof.json")
        open(filename, "r") do io
            MOI.write_to_file(model, io, MOFFile())
        end
    end
end

end
```

Reply:
What about using Julia's pre-existing show/display functions with MIME types (https://docs.julialang.org/en/v1/base/io-network/index.html#Multimedia-I/O-1)?